rbd/mirror_group.go

```diff
@@ -0,0 +1,488 @@
+}
+
+// rbd_mirror_group_resync_fn matches rbd_mirror_group_resync function.
+typedef int(*rbd_mirror_group_resync_fn)(rados_ioctx_t p, const char *name);
+
+// rbd_mirror_group_resync_dlsym take *fn as rbd_mirror_group_resync_fn and calls the dynamically loaded
+// rbd_mirror_group_resync function passed as 1st argument.
+static inline int rbd_mirror_group_resync_dlsym(void *fn, rados_ioctx_t p, const char *name){
+  // cast function pointer fn to rbd_mirror_group_resync and call the function
+  return ((rbd_mirror_group_resync_fn) fn)(p, name);
+}
+
+// rbd_mirror_group_get_info_fn matches rbd_mirror_group_get_info function.
+typedef int(*rbd_mirror_group_get_info_fn)(rados_ioctx_t p, const char *name,
+                                           rbd_mirror_group_info_t *mirror_group_info,
+                                           size_t info_size);
+
+// rbd_mirror_group_get_info_dlsym take *fn as rbd_mirror_group_get_info_fn and calls the dynamically loaded
+// rbd_mirror_group_get_info function passed as 1st argument.
+static inline int rbd_mirror_group_get_info_dlsym(void *fn, rados_ioctx_t p, const char *name,
+                                                  rbd_mirror_group_info_t *mirror_group_info,
+                                                  size_t info_size){
+  // cast function pointer fn to rbd_mirror_group_get_info and call the function
+  return ((rbd_mirror_group_get_info_fn) fn)(p, name, mirror_group_info, info_size);
+}
+
+// rbd_mirror_group_get_global_status_fn matches rbd_mirror_group_get_global_status function.
+typedef int(*rbd_mirror_group_get_global_status_fn)(rados_ioctx_t p, const char *name,
+                                                    rbd_mirror_group_global_status_t *mirror_group_status,
+                                                    size_t status_size);
+
+// rbd_mirror_group_get_global_status_dlsym take *fn as rbd_mirror_group_get_global_status_fn and calls the dynamically loaded
+// rbd_mirror_group_get_global_status function passed as 1st argument.
+static inline int rbd_mirror_group_get_global_status_dlsym(void *fn, rados_ioctx_t p, const char *name,
+                                                           rbd_mirror_group_global_status_t *mirror_group_status,
+														   size_t status_size){
+  // cast function pointer fn to rbd_mirror_group_get_global_status and call the function
+  return ((rbd_mirror_group_get_global_status_fn) fn)(p, name, mirror_group_status, status_size);
+}
+*/
+import "C"
+
+import (
+	"fmt"
+	"unsafe"
+
+	"github.com/ceph/go-ceph/internal/cutil"
+	"github.com/ceph/go-ceph/internal/dlsym"
+	"github.com/ceph/go-ceph/rados"
+)
+
+const (
+	// RBD_SNAP_CREATE_SKIP_QUIESCE flag indicates quiesce'ing is skipped while the group is created
+	RBD_SNAP_CREATE_SKIP_QUIESCE = C.uint32_t(1)
+	// RBD_SNAP_CREATE_IGNORE_QUIESCE_ERROR flag indicates ignore quiesce error while the group is created
+	RBD_SNAP_CREATE_IGNORE_QUIESCE_ERROR = C.uint32_t(2)
```

Comment:
Let's have something similar to 

https://github.com/ceph/go-ceph/blob/d6b48f9b985b8cd5a6d42c816640e18360e75543/rbd/diff_iterate_by_id.go#L12-L15

```
#ifndef RBD_SNAP_CREATE_SKIP_QUIESCE
#define RBD_SNAP_CREATE_SKIP_QUIESCE		(1 << 0) // SHOULD MATCH THE VALUE IN librbd.h
#endif

#ifndef RBD_SNAP_CREATE_IGNORE_QUIESCE_ERROR
#define RBD_SNAP_CREATE_IGNORE_QUIESCE_ERROR	(1 << 1) // SHOULD MATCH THE VALUE IN librbd.h
#endif

```

Reply:
Should we expose these flags to cephcsi ? 

I think these decisions should be made at cephcsi level

Can you explain why specific flags were chosen during different mirror group calls ?